cmd/root.go

```diff
@@ -63,18 +66,45 @@ func init() {
 	rootCmd.PersistentFlags().StringVar(&kubeconfig, "kubeconfig", "", "kubeconfig file path. This is optional, in cluster config will be used if not set")
 }
 
+// See: https://github.com/spf13/viper/issues/188#issuecomment-399884438
+func BindEnvs(iface interface{}, parts ...string) {
+	ifv := reflect.ValueOf(iface)
+	ift := reflect.TypeOf(iface)
+	for i := 0; i < ift.NumField(); i++ {
+		v := ifv.Field(i)
+		t := ift.Field(i)
+		tv, ok := t.Tag.Lookup("mapstructure")
+		if !ok {
+			continue
```

Comment:
If I read the [mapstructure docs] correctly, this is doing the wrong thing: if the tag is missing, it should by default use the field name.  Also, the tag may contain suffixes like `,squash` and `,remain`, so we should strip anything after a comma.

Alternatively, we can actually import `mapstructure`, and convert an instance of `Config` to JSON/YAML and have viper load that before loading the real config, according to a [different comment in that issue].

[mapstructure docs]: https://godoc.org/github.com/mitchellh/mapstructure#hdr-Field_Tags
[different comment in that issue]: https://github.com/spf13/viper/issues/188#issuecomment-413368673

Reply:
> If I read the [mapstructure docs](https://godoc.org/github.com/mitchellh/mapstructure#hdr-Field_Tags) correctly, this is doing the wrong thing: if the tag is missing, it should by default use the field name. Also, the tag may contain suffixes like `,squash` and `,remain`, so we should strip anything after a comma.

Don't see the use case here, and the code would grow too much of complexity for a very tiny improvement that will make it even more fragile. I prefer to look after another way of doing it, avoding reflection at all ( I was unattracted by this solution as well, too ! )

Reply:
I've been playing around with https://github.com/spf13/viper/issues/188#issuecomment-413368673 without luck - it doesn't seem to work anymore, and moreover forces to ship "defaults" which we tried to avoid here, prefering validation of proper input ( to avoid surprises during configuration )